dockerfile_functions.sh

```diff
@@ -89,19 +89,14 @@ print_alpine_pkg() {
 RUN apk --update add --no-cache ca-certificates curl openssl binutils xz \
     && GLIBC_VER="2.25-r0" \
     && ALPINE_GLIBC_REPO="https://github.com/sgerrand/alpine-pkg-glibc/releases/download" \
-    && curl -Ls ${ALPINE_GLIBC_REPO}/${GLIBC_VER}/glibc-${GLIBC_VER}.apk > /tmp/${GLIBC_VER}.apk \
-    && apk add --allow-untrusted /tmp/${GLIBC_VER}.apk \
-    && curl -Ls https://www.archlinux.org/packages/core/x86_64/gcc-libs/download > /tmp/gcc-libs.tar.xz \
-    && mkdir /tmp/gcc \
-    && tar -xf /tmp/gcc-libs.tar.xz -C /tmp/gcc \
-    && mv /tmp/gcc/usr/lib/libgcc* /tmp/gcc/usr/lib/libstdc++* /usr/glibc-compat/lib \
-    && strip /usr/glibc-compat/lib/libgcc_s.so.* /usr/glibc-compat/lib/libstdc++.so* \
-    && curl -Ls https://www.archlinux.org/packages/core/x86_64/zlib/download > /tmp/libz.tar.xz \
-    && mkdir /tmp/libz \
-    && tar -xf /tmp/libz.tar.xz -C /tmp/libz \
-    && mv /tmp/libz/usr/lib/libz.so* /usr/glibc-compat/lib \
+    && curl -Ls https://alpine-pkgs.sgerrand.com/sgerrand.rsa.pub -o /etc/apk/keys/sgerrand.rsa.pub \
+    && curl -Ls ${ALPINE_GLIBC_REPO}/${GLIBC_VER}/glibc-${GLIBC_VER}.apk -o /tmp/${GLIBC_VER}.apk \
+    && apk add /tmp/${GLIBC_VER}.apk \
+    && apk add libgcc \
+    && apk add libstdc++ \
+    && strip /usr/lib/libgcc_s.so.* /usr/lib/libstdc++.so* \
```

Comment:
This wont work. The libgcc, libstc++ and libz that apk installs uses musl libc. Instead, we want the variants of these libraries that use the glibc that you installed in the above step. This is why I was downloading them from archlinux earlier.

Reply:
Why can't we use muscl?

Reply:
Seems OpenJDK is built against glibc and glibc and musl are not always compatible

Reply:
Correct, our current openjdk builds all compile against glibc. musl the default libc in Alpine Linux is not POSIX compliant (It aims for compliance but it not there yet) and it not binary compatible with glibc.